ludwig/data/preprocessing.py

```diff
@@ -940,79 +941,92 @@ def build_dataset(
 
 def build_metadata(dataset_df, features, global_preprocessing_parameters):
     metadata = {}
+
     for feature in features:
-        if PREPROCESSING in feature:
-            preprocessing_parameters = merge_dict(
-                global_preprocessing_parameters[feature[TYPE]],
-                feature[PREPROCESSING]
-            )
-        else:
-            preprocessing_parameters = global_preprocessing_parameters[
-                feature[TYPE]
-            ]
 
-        # deal with encoders that have fixed preprocessing
-        if 'encoder' in feature:
-            encoders_registry = get_from_registry(
-                feature[TYPE],
-                input_type_registry
-            ).encoder_registry
-            encoder_class = encoders_registry[feature['encoder']]
-            if hasattr(encoder_class, 'fixed_preprocessing_parameters'):
-                encoder_fpp = encoder_class.fixed_preprocessing_parameters
+        if HASH not in feature:
+            feature[HASH] = compute_feature_hash(feature)
+
+        if feature[HASH] not in metadata:
```

Comment:
I'm wondering if it would simplify things if `metadata` only dealt with `feature[NAME]` instead of `feature[HASH]`.  We could have a separate map of `feature[HASH]: metadata` so we would not need to recompute.  Then the metadata lookups become much more straightforward (just use `feature[NAME]`).

Reply:
I f this mapping is inside metadata, then it's probably fine, otherwise bringing around a new dictionary will be kinda clunky.

Reply:
No need to bring it along, it would just be for this loop.  So something like:

```
processed_feature_to_metadata = {}
for feature in features:
    if feature[PROC_COLUMN] not in processed_feature_to_metadata:
        processed_feature_to_metadata[feature[PROC_COLUMN]] = ...
    metadata[feature[NAME]] = processed_feature_to_metadata[feature[PROC_COLUMN]]          
```

Then for the rest of the code, the way we access `metadata[feature[NAME]]` doesn't need to change.